ksql-engine/src/main/java/io/confluent/ksql/internal/QueryStateListener.java

```diff
@@ -19,35 +19,34 @@
 import java.util.Collections;
 import java.util.Objects;
 import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.metrics.Gauge;
 import org.apache.kafka.common.metrics.Metrics;
-import org.apache.kafka.streams.KafkaStreams;
 import org.apache.kafka.streams.KafkaStreams.State;
 import org.apache.kafka.streams.KafkaStreams.StateListener;
 
 public class QueryStateListener implements StateListener {
   private final Metrics metrics;
   private final MetricName metricName;
-  private final QueryStateGauge queryStateGauge;
+  private volatile String state = "-";
 
-  public QueryStateListener(
+  QueryStateListener(
       final Metrics metrics,
-      final KafkaStreams kafkaStreams,
-      final String queryApplicationId) {
+      final String queryApplicationId
+  ) {
+    Objects.requireNonNull(queryApplicationId, "queryApplicationId");
     this.metrics = Objects.requireNonNull(metrics, "metrics cannot be null."); ;
```

Comment:
Nit (especially since not a line you changed): there's an extra semicolon at the end of this line.

Reply:
Super nit! ;)